database/migrations/2021_05_13_140700_refactor_size_variants.php

```diff
@@ -0,0 +1,150 @@
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Schema;
+
+class RefactorSizeVariants extends Migration
+{
+	private const PHOTOS_TABLE_NAME = 'photos';
+	private const ID_COL_NAME = 'id';
+	private const SMALL_COL_NAME = 'small';
+	private const SMALL_WIDTH_COL_NAME = 'small_width';
+	private const SMALL_HEIGHT_COL_NAME = 'small_height';
+	private const SMALL2X_COL_NAME = 'small2x';
+	private const SMALL2X_WIDTH_COL_NAME = 'small2x_width';
+	private const SMALL2X_HEIGHT_COL_NAME = 'small2x_height';
+	private const MEDIUM_COL_NAME = 'medium';
+	private const MEDIUM_WIDTH_COL_NAME = 'medium_width';
+	private const MEDIUM_HEIGHT_COL_NAME = 'medium_height';
+	private const MEDIUM2X_COL_NAME = 'medium2x';
+	private const MEDIUM2X_WIDTH_COL_NAME = 'medium2x_width';
+	private const MEDIUM2X_HEIGHT_COL_NAME = 'medium2x_height';
+
+	/**
+	 * Run the migrations.
+	 *
+	 * @return void
+	 */
+	public function up()
+	{
+		Schema::table(self::PHOTOS_TABLE_NAME, function (Blueprint $table) {
+			$table->integer(self::SMALL_WIDTH_COL_NAME)->unsigned()->nullable()->default(null);
+			$table->integer(self::SMALL_HEIGHT_COL_NAME)->unsigned()->nullable()->default(null);
+			$table->integer(self::SMALL2X_WIDTH_COL_NAME)->unsigned()->nullable()->default(null);
+			$table->integer(self::SMALL2X_HEIGHT_COL_NAME)->unsigned()->nullable()->default(null);
+			$table->integer(self::MEDIUM_WIDTH_COL_NAME)->unsigned()->nullable()->default(null);
+			$table->integer(self::MEDIUM_HEIGHT_COL_NAME)->unsigned()->nullable()->default(null);
+			$table->integer(self::MEDIUM2X_WIDTH_COL_NAME)->unsigned()->nullable()->default(null);
+			$table->integer(self::MEDIUM2X_HEIGHT_COL_NAME)->unsigned()->nullable()->default(null);
+		});
+
+		DB::beginTransaction();
+		$photos = DB::table(self::PHOTOS_TABLE_NAME)->select([
+			self::ID_COL_NAME,
+			self::SMALL_COL_NAME,
+			self::SMALL2X_COL_NAME,
+			self::MEDIUM_COL_NAME,
+			self::MEDIUM2X_COL_NAME,
+		])->lazyById();
+
+		foreach ($photos as $photo) {
+			$this->convertUp($photo->{self::SMALL_COL_NAME}, $smallWidth, $smallHeight);
+			$this->convertUp($photo->{self::SMALL2X_COL_NAME}, $small2xWidth, $small2xHeight);
+			$this->convertUp($photo->{self::MEDIUM_COL_NAME}, $mediumWidth, $mediumHeight);
+			$this->convertUp($photo->{self::MEDIUM2X_COL_NAME}, $medium2xWidth, $medium2xHeight);
+
+			DB::table(self::PHOTOS_TABLE_NAME)->where(self::ID_COL_NAME, '=', $photo->id)->update([
+				self::SMALL_WIDTH_COL_NAME => $smallWidth,
+				self::SMALL_HEIGHT_COL_NAME => $smallHeight,
+				self::SMALL2X_WIDTH_COL_NAME => $small2xWidth,
+				self::SMALL2X_HEIGHT_COL_NAME => $small2xHeight,
+				self::MEDIUM_WIDTH_COL_NAME => $mediumWidth,
+				self::MEDIUM_HEIGHT_COL_NAME => $mediumHeight,
+				self::MEDIUM2X_WIDTH_COL_NAME => $medium2xWidth,
+				self::MEDIUM2X_HEIGHT_COL_NAME => $medium2xHeight,
+			]);
+		}
+
+		DB::commit();
+
+		Schema::table(self::PHOTOS_TABLE_NAME, function (Blueprint $table) {
+			$table->dropColumn([
```

Comment:
I don't remember, but I think SQLite does not really like to drop multiple column in one go. I may be wrong. The CI should catch that one anyway.

Reply:
I don't know. My local testing environment uses sqlite and my local development environment uses PostgreSQL. I have run `make test` and `artisan db:migrate` in both environment several times and didn't encounter an error, but may be it depends on the version of sqlite and only a particular version has problems with that. But I can easily fix that in order to be on the save side.

Reply:
The CI will catch it anyway as we do migration and rollback. So if the CI passes, don't bother with it. :)